pkg/controller/lifecycle/actuator.go

```diff
@@ -45,8 +44,10 @@ const (
 )
 
 // NewActuator returns an actuator responsible for Extension resources.
-func NewActuator(config apisconfig.Configuration, chartRendererFactory extensionscontroller.ChartRendererFactory) extension.Actuator {
+func NewActuator(mgr manager.Manager, config apisconfig.Configuration, chartRendererFactory extensionscontroller.ChartRendererFactory) extension.Actuator {
```

Comment:
Same as above.

```suggestion
func NewActuator(client client.Client, decoder runtime.Decoder, config apisconfig.Configuration, chartRendererFactory extensionscontroller.ChartRendererFactory) extension.Actuator {
```

Ref https://github.com/gardener/gardener-extension-registry-cache/blob/f3a362e9eb23c06cff2e2d6e800a395b3433c759/pkg/controller/extension/actuator.go#L43-L50